framework/include/samplers/Sampler.h

```diff
@@ -50,6 +50,24 @@ class Sampler : public MooseObject, public SetupInterface, public DistributionIn
   Sampler(const InputParameters & parameters);
 
   /**
+   * Simple object for storing the sampler location (see SamplerMultiApp).
+   */
+  struct Location
+  {
+    Location(const unsigned int & sample, const unsigned int & row)
+    {
+      this->sample = sample;
+      this->row = row;
+    }
+
+    /// Sample number (i.e., the index in the matrix)
+    unsigned int sample;
+
+    /// Row number for the given sample matrix
+    unsigned int row;
```

Comment:
> Even for a struct that is just a custom location?

We have `struct`s in MOOSE and their member variables start with `_`.

> I would like to avoid having to create methods for retrieving these indicies and also don't want to use foo._sample and foo._row.

If you do not want foo._row, foo._sample, then use methods. Seriously, how much code is this: ``` unsigned int sample() { return _sample; }```.